### PR TITLE
Fix and specifiy payments without source attributes

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Assuming all green this is ready for review.

File: core/app/models/spree/order/checkout.rb [hunk 1]
```diff
@@ -25,16 +25,21 @@ def can_go_to_state?(state)
...
-            source_attributes = checkout_source_attributes(params)
-
             attributes = params.fetch(:order, ActionController::Parameters.new).permit(permitted_params)
 
-            if source_attributes
-              attributes.fetch(:payments_attributes).first.update(
+            payment_attributes = attributes[:payments_attributes].try(:first)
+
+            if payment_attributes
+              updates = {
```
Comment (dkubb) [inline core/app/models/spree/order/checkout.rb]: What about:

```ruby
if payment_attributes
  payment_attributes.update(
    request_env: request_env,
    amount:      total
  )

  source_attributes = checkout_source_attributes(params)
  payment_attributes[:source_attributes] = source_attributes if source_attributes
end

```

**Comment by dkubb** [comment]
@mbj I left only one small comment on this PR.

File: core/app/models/spree/order/checkout.rb (hunk 1 above)
Comment (mbj) [inline core/app/models/spree/order/checkout.rb]: I think that rmoving the immediate lvar is a good idea.

**Comment by mbj** [comment]
@dkubb ready for the next pass

**Pull request merged**